src/sentry/static/sentry/app/stores/guideStore.jsx

```diff
@@ -0,0 +1,74 @@
+import Reflux from 'reflux';
+import GuideActions from '../actions/guideActions';
+
+const GuideStore = Reflux.createStore({
+  init() {
+    this.state = {
+      // All guides returned to us from the server.
+      guides: [],
+      // We record guides seen on the server, but immediately after a user dismisses a guide
+      // it may not have been synced yet, so the local copy helps in filtering correctly.
+      guidesSeen: new Set(),
+      // All anchors that have been registered on this current view.
+      anchors: new Set(),
+      // The "on deck" guide.
+      currentGuide: null,
+      // The current step of the current guide. Null if there's no guide or the guide
+      // is just cued but not opened.
+      currentStep: null,
+    };
+    this.listenTo(GuideActions.fetchSuccess, this.onFetchSuccess);
+    this.listenTo(GuideActions.guideClose, this.onGuideClose);
+    this.listenTo(GuideActions.nextStep, this.onNextStep);
+  },
+
+  onFetchSuccess(data) {
+    this.state.guides = data;
+    this.updateCurrentGuide();
+  },
+
+  onGuideClose() {
+    this.state.guidesSeen.add(this.state.currentGuide.id);
+    this.updateCurrentGuide();
+  },
+
+  onNextStep() {
+    this.state.currentStep =
+      this.state.currentStep === null ? 0 : this.state.currentStep + 1;
+    this.trigger(this.state);
+  },
+
+  updateCurrentGuide() {
+    let available_targets = [...this.state.anchors].map(a => a.props.target);
```

Comment:
just `this.state.anchors` is fine since map doesn't mutate the original array

Reply:
let's use camel case everywhere to keep things consistent?

Reply:
`anchors` is a set

Reply:
👍